continuum/datasets/metashift.py

```diff
@@ -0,0 +1,157 @@
+import pickle as pkl
+import os
+from typing import Iterable, Tuple, Union
+from random import randint, seed
+
+import numpy as np
+
+from continuum import download
+from continuum.datasets.base import _ContinuumDataset
+from continuum.tasks import TaskType
+
+class MetaShift(_ContinuumDataset):
+    # NOTE : using random_contexts can cause some contexts to contain few examples.
+    """Continuum version of the MetaShift dataset.
+
+    References:
+        * MetaShift: A Dataset of Datasets for Evaluating Contextual 
+          Distribution Shifts and Training Conflicts
+          Weixin Liang, James Zou
+          2022
+          arXiv:2202.06523v1
+
+    :param data_path: The folder path containing the data.
+    :param train : 
+    :param download: If true, will download images and the dictionnary 
+           containing classes and contexts if not already in data_path folder.
+    :param train_image_ids: Images ids to use.
+    :param class_names : classes to consider (default = None --> all classes)
+    :param random_context : If true ; only one occurence of each image in a 
+           random choosen class&context combination. If false, all valid contexts are considered, 
+           duplicates of ids will be found in x.
+    :param random_seed : set seed (relevant ponly if random context is True)
+    :param nb_task : set the number of distict tasks.
+    """
+    data_url = "https://nlp.stanford.edu/data/gqa/images.zip"
+    pickle_url = "https://github.com/Weixin-Liang/MetaShift/blob/main/dataset/meta_data/full-candidate-subsets.pkl?raw=true"
+
+    def __init__(
+            self,
+            data_path:str,
+            train:bool = True,
+            download:bool = True,
+            class_names:Union[Iterable[str], None] = None, #Only get images corresponding to specific classe(s)
+            train_image_ids:Union[Iterable[str], None] = None, #Only get specific ids for training. (if None : all)
+            random_contexts:bool = False, #If true, the images will be assigned random class(context) combination among all valid combinations.
+            random_seed:int = 42,
+            nb_tasks:int = 0
```

Comment:
A supplementary option that say if all classes should appear in all tasks would be nice.
with a name like 'strict_domain_inc'.